docs/guides/fullstack_app_guide.md

```diff
@@ -0,0 +1,367 @@
+
+Setting up a server using Flask is easy. You import the package, create the app object, and then create your endpoints. Let's create a basic skeleton for the server first:
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def home():
+    return "Hello World!"
+
+if __name__ == "__main__":
+    app.run(host="0.0.0.0", port=5601)
+```
+
+~flask_demo.py~
+
+If you run this file (`python flask_demo.py`), it will launch a server on port 5601. If you visit `http://localhost:5601/`, you will see the "Hello World!" text rendered in your browser. Nice!
+
+The next step is deciding what functions we want to include in our server, and to start using LlamaIndex.
+
+To keep things simple, the most basic operation we can provide is querying an existing index. Using the [paul graham essay](https://github.com/jerryjliu/llama_index/blob/main/examples/paul_graham_essay/data/paul_graham_essay.txt) from LlamaIndex, create a documents folder and download+place the essay text file inside of it.
+
+### Basic Flask
+
+Now, let's write some code to initialize our index:
+
+```python
+import os
+from llama_index import SimpleDirectoryReader, GPTSimpleVectorIndex
+
+# NOTE: for local testing only, do NOT deploy with your key hardcoded
+os.environ['OPENAI_API_KEY'] = "your key here"
+
+index = None
+
+def initialize_index():
+  global index
+   if os.path.exists(index_name):
+      index = GPTSimpleVectorIndex.load_from_disk(index_name)
+  else:
+      documents = SimpleDirectoryReader("./documents").load_data()
+      index = GPTSimpleVectorIndex(documents)
+      index.save_to_disk(index_name)
+```
+
+This function will initialize our index. If we call this just before starting the flask server in the `main` function, then our index will be ready for user queries!
+
+Our query endpoint will accept `GET` requests with the query text as a parameter. Here's what the full endpoint function will look like:
+
+```python
+from flask import request
+
+@app.route("/query", methods=["GET"])
+def query_index():
+  global index
+  query_text = request.args.get("text", None)
+  if query_text is None:
+    return "No text found, please include a ?text=blah parameter in the URL", 400
+  response = index.query(query_text)
+  return str(response), 200
+```
+
+Now, we've introduced a few new concepts to our server:
+
+- a new `/query` endpoint, defined by the function decorator
+- a new import from flask, `request`, which is used to get parameters from the request
+- if the `text` parameter is missing, then we return an error message and an appropriate HTML response code
+- otherwise, we query the index, and return the response as a string
+
+A full query example that you can test in your browser might look something like this: `http://localhost:5601/query?text=what did the author do growing up`

```

Comment:
in the browser the strings will be escaped (though i'm sure you know this), maybe add a parentheses to clarify?